### PR TITLE
Add profile to cross compile for intel on macOs

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

As m1/m2 becomes more used we should also have a profile to cross compile for intel.

Modifications:

- Add profile to cross-compile for intel when running on m1/m2.
- Adjust finish_release.sh script to use correct cross-compile profile based on the architecture.

Result:

Be able to cross compile for intel and also running the release on m1/m2

### CODE REVIEW

File: boringssl-static/pom.xml [hunk 1]
```diff
@@ -1113,6 +1113,254 @@
...
+        <cmakeCxxFlags>-O3 -fno-omit-frame-pointer -target ${target} -Wno-error=range-loop-analysis</cmakeCxxFlags>
+        <cflags>-target ${target} -Werror -fno-omit-frame-pointer -fvisibility=hidden -Wunused -Wno-unused-value -O3</cflags>
+        <cppflags>-target ${target} -DHAVE_OPENSSL -I${boringsslSourceDir}/include</cppflags>
+        <ldflags>-arch arm64 -L${boringsslHome}/ssl -L${boringsslHome}/crypto -lssl -lcrypto</ldflags>
+        <!-- use aarch_64 as this is also what os.detected.arch will use on an aarch64 system -->
+        <nativeJarFile>${project.build.directory}/${project.build.finalName}-${os.detected.name}-osx_64.jar</nativeJarFile>
+        <nativeLibOsParts>${os.detected.name}_osx_64</nativeLibOsParts>
+        <jniClassifier>${os.detected.name}-osx_64</jniClassifier>
+        <jniArch>osx_64</jniArch>
+        <javaModuleNameClassifier>${os.detected.name}.osx_64</javaModuleNameClassifier>
```
Comment (chrisvest) [inline boringssl-static/pom.xml:1140]: What is the `osx_64` part used in these lines? Should it be `osx_x86_64` or something like that?

File: scripts/finish_release.sh [hunk 2]
```diff
@@ -20,13 +20,19 @@ if git tag | grep -q "$2" ; then
     exit 1
 fi
 
+CROSS_COMPILE_PROFILE="mac-m1-cross-compile"
+if [ "$ARCH" == "arm" ]; then
```
Comment (chrisvest) [inline scripts/finish_release.sh:24]: On my machine, `$ARCH` is not defined. I do, however, have the following:

```
% uname -p
arm
```

File: scripts/finish_release.sh [hunk 3]
```diff
@@ -20,13 +20,19 @@ if git tag | grep -q "$2" ; then
...
+    PROFILE="mac-intel-cross-compile"
```
Comment (chrisvest) [inline scripts/finish_release.sh:25]: The variable assigned here should be `CROSS_COMPILE_PROFILE`

**Review by chrisvest — CHANGES_REQUESTED** [review]

File: boringssl-static/pom.xml (hunk 1 above)
Comment (normanmaurer) [reply boringssl-static/pom.xml]: typo

File: scripts/finish_release.sh (hunk 2 above)
Comment (normanmaurer) [reply scripts/finish_release.sh:25]: missed to copy some stuff... fixed

**Review by chrisvest — APPROVED** [review]

**Pull request merged**